src/test/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGeneratorTest.java

```diff
@@ -458,4 +460,28 @@ public void testConstructorWithTreeWalkerAuditEvent() {
         assertEquals(expected, actual, "Generated queries do not match expected ones");
     }
 
+    @Test
+    public void testEncode() {
+        final Map<String, String> encoding = new HashMap<>();
+
+        encoding.put("<", "&lt;");
+        encoding.put(">", "&gt;");
+        encoding.put("'", "&apos;&apos;");
+        encoding.put("\"", "&quot;");
+        encoding.put("&", "&amp;");
+        encoding.put("&lt;", "&amp;lt;");
+        encoding.put("abc;", "abc;");
+        encoding.put("&#0;", "&amp;#0;");
+        encoding.put("&#0", "&amp;#0");
+        encoding.put("&#X0;", "&amp;#X0;");
+        encoding.put("\u0001", "#x1;");
+        encoding.put("\u0080", "#x80;");
+        encoding.put("\n", "&#10;");
+        encoding.put("\r", "");
+
+        for (Map.Entry<String, String> put : encoding.entrySet()) {
+            final String encoded = XpathQueryGenerator.encode(put.getKey());
+            assertEquals(put.getValue(), encoded, "\"" + put.getKey() + "\"");
```

Comment:
Please use `truth.assertWithMessage("...").that(...)` here.

Reply:
@pbludov I changed `assertEquals(put.getValue(), encoded, "\"" + put.getKey() + "\"")` -> `Truth.assertWithMessage("Incorrect encoding").that(entry.getValue()).isEqualTo(encoded)` but it is giving me pmd error of ` Rule:JUnitTestsShouldIncludeAssert Priority:3 JUnit tests should include assert() or fail().` So I had to include both assertEquals and Truth test test. Is it Ok?

Reply:
Just use static import. See this [example](https://github.com/checkstyle/checkstyle/blob/1e3fb3fae940bf7ea340592f7f0cda3af1b320e3/src/test/java/com/puppycrawl/tools/checkstyle/utils/XpathUtilTest.java#L178)

Reply:
Done